Sources/Subs.php

```diff
@@ -2870,6 +2864,37 @@ function highlight_php_code($code)
 	return strtr($buffer, array('\'' => '&#039;', '<code>' => '', '</code>' => ''));
 }
 
+/**
+ * Gets the appropriate URL to use for images (or whatever) when using SSL
+ *
+ * The returned URL may or may not be a proxied URL, depending on the situation.
+ * Mods can implement alternative proxies using the 'integrate_proxy' hook.
+ *
+ * @param string $url The original URL of the requested resource
+ * @return string The URL to use
+ */
+function get_proxied_url($url)
+{
+	global $boardurl, $modSettings, $image_proxy_enabled, $image_proxy_secret;
+
+	// Only use the proxy if enabled and necessary
+	if (empty($image_proxy_enabled) || empty($modSettings['force_ssl']) || parse_url($url, PHP_URL_SCHEME) === 'https')
```

Comment:
In the load php is already a check for force_ssl,
when disable it set $image_proxy_enabled at false:
https://github.com/SimpleMachines/SMF2.1/blob/1bbcf2bf43d82a61cb1bb94533e8b15a0a5247f4/Sources/Load.php#L1759-L1761
no need for double check empty($modSettings['force_ssl'])

Reply:
Good catch.